### PR TITLE
Databases and roles updates

### PR DESCRIPTION
**Pull request by danieltprice**
**First Draft** - this will likely need to evolve to ensure it's covered adequately. I updated the database and roles docs, and I added a "Manage database access with SQL" guide that covers it all.

Previews:
- "The neon_superuser role"
-https://neon-next-git-dprice-databases-roles-sql-neondatabase.vercel.app/docs/manage/roles#the-neonsuperuser-role
- "Manage databases with SQL"
https://neon-next-git-dprice-databases-roles-sql-neondatabase.vercel.app/docs/manage/databases#manage-databases-with-sql
- "Manage roles with SQL"
https://neon-next-git-dprice-databases-roles-sql-neondatabase.vercel.app/docs/manage/roles#manage-roles-with-sql
- "Manage database access with SQL" guide:
https://neon-next-git-dprice-databases-roles-sql-neondatabase.vercel.app/docs/guides/manage-database-access

### CODE REVIEW

File: content/docs/guides/manage-database-access.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+title: Manage roles and database access with SQL
+subtitle: Learn how to create roles and manage database access in Neon with SQL
+enableTableOfContents: true
+---
+
+This guide shows how to manage database access in Neon using SQL. This guide will lead you through connecting to Neon with an administrative role, creating a database, creating a new role for privilege management, and granting privileges to that role. It will then show how to create roles for database users and grant those users role membership that will allow them to use the new database.
+
+## Understanding roles in Neon
+
+Before you begin, it's important to understand how roles work in Neon. Each Neon project is created with a default role that takes its name from your Neon account (the Google, GitHub, or partner account that you registered with). This role owns the default database (`neondb`) that is created in your project's primary branch. For example, if I sign up for Neon with a John Smith Google account, the the project is created with a default role named `john`.
```
Comment (m-abdelwahab) [inline content/docs/guides/manage-database-access.md:11]:
```suggestion
Before you begin, it's important to understand how roles work in Neon. Each Neon project is created with a default role that takes its name from your Neon account (the Google, GitHub, or partner account that you registered with). This role owns the default database (`neondb`) that is created in your project's primary branch. For example, if you sign up for Neon with a John Smith Google account, the the project is created with a default role named `john`.
```

File: content/docs/guides/manage-database-access.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+    ```sql
+    CREATE DATABASE app_db;
+    ```
+
+    <Admonition type="note">
+    The role that creates a database is automatically the owner of the database.
+    </Admonition>
+
+3. Create a 'group' role for database users. This role will be used to manage database user privileges. Neon requires a password when creating any role with SQL.
```
Comment (m-abdelwahab) [inline content/docs/guides/manage-database-access.md:55]:
```suggestion
3. Create a shared role for database users. This role will be used to manage database user privileges. Neon requires a password when creating any role with SQL.
```

When reading `'group'`, I thought I'm going to create a role called `group`

File: content/docs/guides/manage-database-access.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    ```
+
+    <Admonition type="note">
+    The role that creates a database is automatically the owner of the database.
+    </Admonition>
+
+3. Create a shared role for database users. This role will be used to manage database user privileges. Neon requires a password when creating any role with SQL.
+
+    ```sql
+    CREATE ROLE dev_users PASSWORD `password`;
```
Comment (vadim2404) [inline content/docs/guides/manage-database-access.md:58]: it is worth mentioning that we do not support creating users with hashed passwords, like

```sql
CREATE USER user_name
WITH PASSWORD 'SCRAM-SHA-256$4096:UunGvPETiX/JNGBvjOgW9A==$CPGNh7/MRfs0ispH9/HSJajOI8Uhp+UCRo/b/ToXIEY=:L6NzxQ3XUeWEeRa+oiuajC9Vgl7wk6ZpHAHl+pv4m00=';
```

**Comment by m-abdelwahab** [comment]
@danieltprice One term that is often used is "root user". Perhaps we should include it somehow so that it shows up in search?

File: content/docs/guides/manage-database-access.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+    ```sql
+    GRANT dev_users TO dev_user1;
+    GRANT dev_users TO dev_user2;
+    ```
+
+    The `dev_user1` and `dev_user2` can now connect to the `app_db` database and start using it with full privileges.
+
+    ```bash
+    psql postgres://dev_user1:<password>@ep-restless-waterfall-733645.us-west-2.aws.neon.tech/app_db
```
[the remark this answers is not in the document]
Comment (danieltprice) [reply content/docs/guides/manage-database-access.md:100]: Used the example password instead.

File: content/docs/guides/manage-database-access.md (hunk 3 above)
Comment (danieltprice) [reply content/docs/guides/manage-database-access.md:58]: Added "Hashed passwords are not supported" and that passwords are required.

**Comment by danieltprice** [comment]
> @danieltprice One term that is often used is "root user". Perhaps we should include it somehow so that it shows up in search?

Added this note:

"Neon is a managed PostgreSQL service, so you cannot access the host operating system, and you can't connect using the PostgreSQL `superuser` account like you can with a stand-alone PostgreSQL installation."

There's no concept of "root" in Postgres, so I'm not sure about mentioning it in the docs, but I will add a synonym to Algolia.

**Review by m-abdelwahab — APPROVED** [review]

**Pull request merged**